tls/s2n_signature_algorithms.c

```diff
@@ -92,6 +92,35 @@ static int s2n_choose_sig_scheme(struct s2n_connection *conn, struct s2n_sig_sch
     S2N_ERROR(S2N_ERR_INVALID_SIGNATURE_SCHEME);
 }
 
+/* exactly like s2n_choose_sig_scheme() without matching client's preference */
+static int s2n_tls13_preferred_sig_scheme(struct s2n_connection *conn, struct s2n_signature_scheme *chosen_scheme_out)
+{
+    notnull_check(conn);
+    const struct s2n_signature_preferences *signature_preferences = NULL;
+    GUARD(s2n_connection_get_signature_preferences(conn, &signature_preferences));
+    notnull_check(signature_preferences);
+
+    struct s2n_cipher_suite *cipher_suite = conn->secure.cipher_suite;
+    notnull_check(cipher_suite);
+
+    for (size_t i = 0; i < signature_preferences->count; i++) {
+        const struct s2n_signature_scheme *candidate = signature_preferences->signature_schemes[i];
+
+        if (s2n_signature_scheme_valid_to_accept(conn, candidate) != S2N_SUCCESS) {
+            continue;
+        }
+
+        if (s2n_is_sig_scheme_valid_for_auth(conn, candidate) != S2N_SUCCESS) {
+            continue;
+        }
```

Comment:
This is duplicating the logic from s2n_choose_sig_scheme. If we add more rules about choosing signature schemes, will have to make sure we add them in both places now. Maybe combine these two checks into a new helper method? Or turn our own preference list into a s2n_sig_scheme_list so that you can call s2n_choose_sig_scheme?